nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/VisibilityFetchSupport.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.hbase;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public interface VisibilityFetchSupport {
+    PropertyDescriptor AUTHORIZATIONS = new PropertyDescriptor.Builder()
+        .name("hbase-fetch-row-authorizations")
+        .displayName("Authorizations")
+        .description("The list of authorizations to pass to the scanner. This will be ignored if cell visibility labels are not in use.")
```

Comment:
The labels are used for authorizations certainly, but they're used to filter specific cells, too, right? E.g. if tag 'A' and tag 'B' are associated with a user, then the user can access following cells:
- Tagged with 'A'
- Tagged with 'A' and 'B'
- Tagged with 'B'

In that case, if NiFi user would like to fetch/scan only cells tagged with 'B', the name of the property 'Authorizations' does not sound perfect to me. It would make more sense if we called it simply 'Visibility Labels'. How do you think?

Reply:
In your three cases, the visibility labels would be:

* A
* A&B
* B

Passing `[B]` would eliminate the first two.

With HBase and Accumulo, you can tell the scanner "yeah, I know have set X, but use subset Y of my authorizations" because there are so many valid use cases where the scanner has to limit itself on the authorizations. A good example is that in a user-facing deployment, the scanner user will frequently have all authorizations, but be configured on the fly to do a scan with those of a particular user sitting at the keyboard setting up a scan.

Again, the distinction between "authorizations" and labels" holds because label is the requirement statement and authorizations are the tokens used to hopefully make it evaluate to `true`.

Reply:
> Do you still think 'Authorizations' is the better naming??

I do because we really need to draw a firm distinction between the label that is being written to the cell's metadata and the list of privileges.

Reply:
I agree on drawing a firm distinction between those, I think it just a matter of naming. Especially it was ambiguous to me where I am supposed to sed 'A set of Visibility Labels' meaning just a comma separated list of labels and 'Visibility Expression', a logical formula to verify users accessibility.

I will comment on processor properties to propose more specific suggestions later. Thanks.